gato/training/trainer.py

```diff
@@ -105,9 +108,13 @@ def train_iteration(self, num_steps, iter):
     def train_step(self):
         logs = {}
         logs['training/learning_rate'] = self.scheduler.get_lr()[0] # store LR at current step
-        
+
         # Build training batch
-        batch_dicts = self.sample_control_batch(self.args.batch_size)
+        # for simplicity sake - assume interleave training on control & text
+        if self.steps % 2 == 0:
```

Comment:
I think we should try to implementing mixing both text/control in one batch. Maybe have a configurable argument like text_prop(value from 0 to 1).  Sample texst_prop * batch_size sequences of text, say control_prop = 1-text_prop for now, and control batch size is control_prop * batch_size. Sample sub batches of these sizes, join them together into one batch, and take a gradient up using this combined batch. We may want to restructure tokenize_input_dicts and forward() to potentially make this easier, can elaborate on a separate comment.

Reply:
Resolved in latest commit.